src/Query/Query.php

```diff
@@ -694,6 +695,17 @@ public function union(QueryInterface|string $sql, bool $all = false): static
     }
 
     public function where(array|string|ExpressionInterface|null $condition, array $params = []): static
+    {
+        if ($this->where === null) {
+            $this->where = $condition;
+        } else {
+            throw new LogicException('The `where` condition was set earlier. Use the `setWhere()`, `andWhere()` or `orWhere()` method.');
+        }
+        $this->addParams($params);
+        return $this;
+    }
+
+    public function setWhere(array|string|ExpressionInterface|null $condition, array $params = []): static
```

Comment:
I'd suggest to have `clearWhere` / `dropWhere` to keep assigning logic in `where` method only. 
4  `where` methods seems confusing for me

Reply:
@Tigrov nope, you should compare these:
```
$query->where($condition)->all();
$query->setWhere($condition)->all();
```

`reset` method is used only for cases when you want to clear without further overriding:

```
$query->resetWhere()->all();
$query->setWhere(null)->all();
```

in this case `reset` looks better, isn't it?

Reply:
This is just a special case. We can make the default value in `setWhere()` equals to `null`